src/utils/communication_buffer.hpp

```diff
@@ -93,13 +101,19 @@ class CommBuffer {
   T &buffer() { return buf_; }
   const T &buffer() const { return buf_; }
 
-  void Allocate() {
-    if (!active_) {
-      buf_ = get_resource_();
+  void Allocate(int size = -1) {
+    if (!active_ || (size > 0 && buf_.size() != size)) {
+      buf_ = get_resource_(size);
```

Comment:
Is there a danger that `get_resource_` is now called with `-1` as argument?

Reply:
Actually, this does seem to be the case with the 
```
   // For early start of Irecv, always need storage space even if not used
    if (!IsActive()) Allocate(-1);
```
call below.
So what does this met calling `get_resource_` with `-1`?

Reply:
See answer to question below.